docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,90 @@
+Log Centralisation
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workload running on top
+of it.
+
+MetalK8s provides monitoring service with Prometheus and AlertManager in
+order to expose system performance time series and alerts.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information that can be very helpful when it comes to understand the
+root cause of a failure or a performance degradation. Because of the
+distributed nature of MetalK8s and workloads running of top of it, it is
+important to have some tooling to collect and aggregate all logs in order to
+make them browsable from a central endpoint. It is also important to ease
+browsing and correlation with other type of indicators such as KPIs and alerts.
+
+In addition to the application logs, the platform and workload are also
+producing some audit logs. If the same system is used to manage application and
+audit logs, then it is important to provide the end users of the workloads
+some capabilities to secure and query the audit logs belonging to their usage.
+
+Requirements
+------------
+
+- Collect both containers and system logs (STDOUT or STDERR)
+- Persistent storage of logs
```

Comment:
For how long? Also, there's no mention of "downsampling" (doesn't make much sense for logs per-se, but extracting long-term metrics out of logs can be useful data).

Reply:
I don't know yet but it has to be configurable anyway. This should be part of CSC and centralized CLI..
I see some value in extracting some KPIs like nbr of occurrences or frequency of a certain type of log, do you think about something else?

Reply:
It needs to be configurable, but we need sensible defaults that come from "real data". That should be part of the requirements I believe (defining the sensible defaults based on various architectures).

> nbr of occurrences or frequency of a certain type of log, do you think about something else?
Not really, except if the log itself is outputting numeric values (e.g. an old component without an exporter could just log the duration of an operation after completion).